wormhole-connect/src/utils/inProgressTxCache.ts

```diff
@@ -9,6 +9,89 @@ const LOCAL_STORAGE_MAX = 3;
 const JSONReplacer = (_, value: any) =>
   typeof value === 'bigint' ? value.toString() : value;
 
+// Checks the existance of the given prop in a parent object
+const validateChildProp = (
+  parent: object,
+  propName: string,
+  propType: string,
+) => {
+  return (
+    propName in parent &&
+    parent[propName] !== null && // Line below is an explicit check for undefined values as well
+    typeof parent[propName] === propType
+  );
+};
+
+// Validates a single local transaction
+const validateSingleTransaction = (
+  tx: TransactionLocal,
+): tx is TransactionLocal => {
+  // Check first level required properties
+  const hasReceipt = validateChildProp(tx, 'receipt', 'object');
+  const hasRoute = validateChildProp(tx, 'route', 'string');
+  const hasTimestamp = validateChildProp(tx, 'timestamp', 'number');
+  const hasTxDetails = validateChildProp(tx, 'txDetails', 'object');
+  const hasTxHash = validateChildProp(tx, 'txHash', 'string');
+
+  if (
+    !hasReceipt ||
```

Comment:
A more performant but less reader-friendly way would be to use the function call here (e.g. `!validateChildProp(tx, 'receipt', 'object') ||`), so when one evals truty, it won't run the rest.

Reply:
you could refactor slightly and get that benefit while making it even more readable imo. change `validateChildProp` to plural `validateChildProps` (or `validateChildTypes`) and accept an object which represents the spec:

```ts
validateChildProps(tx, {
  receipt: 'object',
  route: 'string',
  timestamp: 'number',
  ...
});
```

this func would iterate through the spec, and short circuit with `return false` as soon as it finds a mismatch.